arkouda/numeric.py

```diff
@@ -468,30 +481,38 @@ def hash(
     fixed key for the hash, which makes it possible for an
     adversary with control over input to engineer collisions.
 
-    In the case of a list of pdrrays being passed, a non-linear
-    function must be applied to each array since hashes of subsequent
-    arrays cannot be simply XORed because equivalent values will
-    cancel each other out, hence we do a rotation by the ordinal of
-    the array.
+    In the case of a list of pdrrays, Strings, or Segarrays
+    being passed, a non-linear function must be applied to each
+    array since hashes of subsequent arrays cannot be simply XORed
+    because equivalent values will cancel each other out, hence we
+    do a rotation by the ordinal of the array.
     """
-    if isinstance(pda, pdarray):
-        return _hash_single(pda, full)
-
-    repMsg = type_cast(
-        str,
-        generic_msg(
-            cmd="efuncArr",
-            args={
-                "nameslist": [n.name for n in pda],
-                "typeslist": [n.objType for n in pda],
-                "length": len(pda),
-                "size": len(pda[0]),
-            },
-        ),
-    )
-
-    a, b = repMsg.split("+")
-    return create_pdarray(a), create_pdarray(b)
+    from arkouda import SegArray as _Segarray
```

Comment:
Just noting again that we may want switch this to `SegArray_` to align with our existing cases. Though I would be open to adding an issue to change everything to this format moving forward as well.